src/core/init/stream_events_emitter/index.ts

```diff
@@ -0,0 +1,178 @@
+/**
```

Comment:
Also, I didn't see it but we try to not put code in an `index.ts` anymore, just export statements.

Here it should probably be:
 `src/core/init/stream_events_emitter/stream_events_emitter.ts`
or even could be something like
`src/core/stream_events/stream_events_emitter.ts`